nx/lib/nx/defn/expr.ex

```diff
@@ -439,13 +439,9 @@ defmodule Nx.Defn.Expr do
                 description: """
                 cond/if expects all branches to return compatible tensor types.
 
-                Got mismatching templates:
+                Comparison:
 
-                #{inspect_as_template(first)}
-
-                and
-
-                #{inspect_as_template(expr)}
+                #{Nx.Defn.CompilationDiff.build_and_inspect(first, expr)}
```

Comment:
This will be wrong for cond. cond actually allows different shapes and sizes, but not different containers. You need to pass a function that tells you how to compare tensors!

Reply:
Ah right, just like above, right?